torch/csrc/jit/runtime/vararg_functions.cpp

```diff
@@ -39,6 +39,108 @@ void format(Stack& stack, size_t num_inputs) {
   push(stack, ss.str());
 }
 
+// IValue tags are intentionally private, so we need additional logic to cast
+// the IValue type to the specified format.
+std::string getFormattedArg(char key, const IValue& ival, int precision=6) {
+  //TODO: Implement precison-based formatting
+  std::stringstream ss;
+  bool added = false;
+  switch (key) {
+    case 'd':
+    case 'i': {
+      TORCH_CHECK(
+          ival.isScalar(),
+          "%",
+          key,
+          " format: A number is required, not ",
+          ival.tagKind());
+      if (ival.isDouble()) {
+        std::stringstream().swap(ss);
+        ss << static_cast<int>(ival.toDouble());
+        added = true;
+      }
+      break;
+    }
+    case 'e':
+    case 'E': {
+      TORCH_CHECK(
+        ival.isScalar(),
+        "%",
+        key,
+        " format: A number is required, not ",
```

Comment:
Conventionally, error message should be something like
XXX is required *got* YYY

Reply:
Thanks, I'll change it! I was copying Python's error message.